Mathlib/Data/Finset/NatAntidiagonal.lean

```diff
@@ -126,6 +126,14 @@ theorem filter_snd_eq_antidiagonal (n m : ℕ) :
   simp [filter_map, this, filter_fst_eq_antidiagonal, apply_ite (Finset.map _)]
 #align finset.nat.filter_snd_eq_antidiagonal Finset.Nat.filter_snd_eq_antidiagonal
 
+@[simp] lemma antidiagonal_filter_le_fst {n : ℕ} :
+    (antidiagonal n).filter (fun a ↦ n ≤ a.fst) = {(n, 0)} := by
+  ext; aesop
```

Comment:
There's a more general result here, that `(antidiagonal n).filter (fun a ↦ k ≤ a.fst)` is `(antidiagonal (n - k)).map (Function.Embedding.prodMap sorry Embedding.refl)`; can you add that one instead, since I think your result then follows by `simp`.

Reply:
Neat, done!

I also confirm that my original lemmas do now follow by `simp`.